### ISSUE TITLE
Ruby bindings are not compatible with Ruby 3.0

### ISSUE THREAD

**Issue by ryanwilsonperkin** [cloudsmith-io/cloudsmith-api#24]
The ruby bindings [make use](https://github.com/cloudsmith-io/cloudsmith-api/blob/master/bindings/ruby/src/lib/cloudsmith-api/configuration.rb#L178) of the `URI.encode` method which was deprecated in Ruby 2.5 and has been removed in Ruby 3.0

This causes an issue when we attempt to make any API call with the Cloudsmith-API gem.

This appears to have been addressed in swagger-codegen in https://github.com/swagger-api/swagger-codegen/pull/10445

I've validated this with the following minimal reproduction:

```sh
> ruby --version
ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]

> gem info cloudsmith-api

*** LOCAL GEMS ***

cloudsmith-api (0.57.1)
    Author: Cloudsmith Ltd
    Homepage: https://api.cloudsmith.io/?format&#x3D;openapi
    License: Apache-2.0
    Installed at: /home/ryan/.bundle/cloudsmith-example

    Cloudsmith API

> cat cloudsmith-example.rb
require 'cloudsmith-api'

CloudsmithApi.configure do |config|
  config.api_key['X-Api-Key'] = 'my-example-api-key'
end

CloudsmithApi::PackagesApi.new.packages_list(
  "example-organization",
  "gems",
  query: "name:example"
)

> ruby cloudsmith-example.rb
/home/ryan/.bundle/cloudsmith-example/gems/cloudsmith-api-0.57.1/lib/cloudsmith-api/configuration.rb:178:in `base_url': undefined method `encode' for URI:Module (NoMethodError)
        from /home/ryan/.bundle/cloudsmith-example/gems/cloudsmith-api-0.57.1/lib/cloudsmith-api/api_client.rb:267:in `build_request_url'
        from /home/ryan/.bundle/cloudsmith-example/gems/cloudsmith-api-0.57.1/lib/cloudsmith-api/api_client.rb:90:in `build_request'
        from /home/ryan/.bundle/cloudsmith-example/gems/cloudsmith-api-0.57.1/lib/cloudsmith-api/api_client.rb:50:in `call_api'
        from /home/ryan/.bundle/cloudsmith-example/gems/cloudsmith-api-0.57.1/lib/cloudsmith-api/api/packages_api.rb:205:in `packages_list_with_http_info'
        from /home/ryan/.bundle/cloudsmith-example/gems/cloudsmith-api-0.57.1/lib/cloudsmith-api/api/packages_api.rb:162:in `packages_list'
        from cloudsmith-example.rb:7:in `<main>'
```

**Comment by ryanwilsonperkin** [comment]
Upgrading to at least v2.4.16 of swagger-codegen and re-running codegen should resolve this issue.

**Comment by paddycarey** [comment]
Thanks for reporting @ryanwilsonperkin 

should be closed by #27, let me know if you have any issues!
